### PR TITLE
use office-fabric-ui components

### PR DESCRIPTION
**Pull request by lvybriage**
This PR‘s content:

- use `office-fabric-ui` , delete `antd` dependency
- refactor nav bar
- refactor `addColumn` modal

improvement

- table： 击selecte icon可以选中一行但不需要打开expand row [调试中，有bug, 期待bug bash进]
- trial's intermediate result graph: x轴名称改为trial sequence id

support items in next release:

- Parameters | Log in open row: div保持一样的高度
- tableList pagination function
- 单独的refresh button function

### CODE REVIEW

File: src/webui/src/components/trial-detail/TableList.tsx [hunk 1]
```diff
@@ -373,136 +388,132 @@ class TableList extends React.Component<TableListProps, TableListState> {
                 parameterStr.push(`${value} (search space)`);
             });
         }
+        let showTitle = COLUMNPro; // eslint-disable-line @typescript-eslint/no-unused-vars
         showTitle = COLUMNPro.concat(parameterStr);
 
         // only succeed trials have final keys
         if (tableSource.filter(record => record.status === 'SUCCEEDED').length >= 1) {
             const temp = tableSource.filter(record => record.status === 'SUCCEEDED')[0].accuracy;
```
Comment (QuanluZhang) [inline src/webui/src/components/trial-detail/TableList.tsx]: `temp`: better to give it a meaningful name.

File: src/webui/src/static/const.ts [hunk 2]
```diff
@@ -2,8 +2,9 @@
 const METRIC_GROUP_UPDATE_THRESHOLD = 100;
 const METRIC_GROUP_UPDATE_SIZE = 20;
 
-const MANAGER_IP = `/api/v1/nni`;
+const MANAGER_IP = `http://13.77.78.63:8080/api/v1/nni`;
```
Comment (QuanluZhang) [inline src/webui/src/static/const.ts]: remove this ip address?

File: src/webui/config/webpackDevServer.config.js [hunk 3]
```diff
@@ -59,36 +60,44 @@ module.exports = function(proxy, allowedHost) {
...
     // src/node_modules is not ignored to support absolute imports
-    // https://github.com/facebookincubator/create-react-app/issues/1065
+    // https://github.com/facebook/create-react-app/issues/1065
     watchOptions: {
       ignored: ignoredFiles(paths.appSrc),
     },
     // Enable HTTPS if the HTTPS environment variable is set to 'true'
     https: protocol === 'https',
-    host: host,
+    host,
```
Comment (SparkSnail) [inline src/webui/config/webpackDevServer.config.js]: why remove `host:`

File: src/webui/scripts/build.js [hunk 4]
```diff
@@ -95,22 +108,49 @@ measureFileSizesBeforeBuild(paths.appBuild)
...
+  // We used to support resolving modules according to `NODE_PATH`.
+  // This now has been deprecated in favor of jsconfig/tsconfig.json
+  // This lets you use absolute paths in imports inside large monorepos:
+  if (process.env.NODE_PATH) {
+    console.log(
+      chalk.yellow(
+        'Setting NODE_PATH to resolve modules absolutely has been deprecated in favor of setting baseUrl in jsconfig.json (or tsconfig.json if you are using TypeScript) and will be removed in a future major release of create-react-app.'
+      )
+    );
+    console.log();
```
Comment (SparkSnail) [inline src/webui/scripts/build.js]: empty content

File: src/webui/scripts/start.js [hunk 5]
```diff
@@ -54,35 +55,51 @@ if (process.env.HOST) {
   console.log(
     `If this was unintentional, check that you haven't mistakenly set it in your shell.`
   );
-  console.log(`Learn more here: ${chalk.yellow('http://bit.ly/2mwWSwH')}`);
+  console.log(
+    `Learn more here: ${chalk.yellow('https://bit.ly/CRA-advanced-config')}`
+  );
   console.log();
```
Comment (SparkSnail) [inline src/webui/scripts/start.js]: empty content

File: src/webui/src/components/trial-detail/Intermediate.tsx [hunk 6]
```diff
@@ -226,62 +231,64 @@ class Intermediate extends React.Component<IntermediateProps, IntermediateState>
...
+    //     if (whichGraph === '4') {
+    //         const { isFilter, length, clickCounts } = nextState;
+    //         const beforeLength = this.state.length;
+    //         const beforeSource = this.props.source;
+    //         const beforeClickCounts = this.state.clickCounts;
 
-            if (isFilter !== this.state.isFilter) {
-                return true;
-            }
+    //         if (isFilter !== this.state.isFilter) {
```
Comment (SparkSnail) [inline src/webui/src/components/trial-detail/Intermediate.tsx]: why annotate these lines?

File: src/webui/src/components/trial-detail/Para.tsx [hunk 7]
```diff
@@ -571,89 +591,83 @@ class Para extends React.Component<ParaProps, ParaState> {
...
-    shouldComponentUpdate(nextProps: ParaProps, nextState: ParaState): boolean {
+    // shouldComponentUpdate(nextProps: ParaProps, nextState: ParaState): boolean {
 
-        const { whichGraph } = nextProps;
-        const beforeGraph = this.props.whichGraph;
-        if (whichGraph === '2') {
-            if (whichGraph !== beforeGraph) {
-                return true;
-            }
+    //     const { whichGraph } = nextProps;
```
Comment (SparkSnail) [inline src/webui/src/components/trial-detail/Para.tsx]: If these lines are not useful, could remove them.

File: src/webui/src/static/function.ts [hunk 8]
```diff
@@ -118,33 +117,36 @@ const killJob = (key: number, id: string, status: string, updateList?: Function)
...
+                // message.destroy();
+                // message.success('Cancel the job successfully');
                 // render the table
                 if (updateList) {
                     updateList();  // FIXME
                 }
             } else {
-                message.error('fail to cancel the job');
+                alert('fail to cancel the job');
+                // message.error('fail to cancel the job');
```
Comment (SparkSnail) [inline src/webui/src/static/function.ts]: remove unused function?

File: src/webui/src/components/trial-detail/TableList.tsx (hunk 1 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/TableList.tsx]: Ok, thanks

File: src/webui/config/webpackDevServer.config.js (hunk 3 above)
Comment (lvybriage) [reply src/webui/config/webpackDevServer.config.js]: 同名可以简写~ 这是项目生成时的文件，我其实并没有改什么...

File: src/webui/scripts/build.js (hunk 4 above)
Comment (lvybriage) [reply src/webui/scripts/build.js]: I had delete this line.

File: src/webui/scripts/start.js (hunk 5 above)
Comment (lvybriage) [reply src/webui/scripts/start.js]: Thanks for reviewing! I had delete this line.

File: src/webui/src/components/trial-detail/Intermediate.tsx (hunk 6 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/Intermediate.tsx]: // will add it later. [graph is not render when add this component]
So... will delete this notes. will support this function after that.

File: src/webui/src/components/trial-detail/Para.tsx (hunk 7 above)
Comment (lvybriage) [reply src/webui/src/components/trial-detail/Para.tsx]: Ok.

File: src/webui/src/static/function.ts (hunk 8 above)
Comment (lvybriage) [reply src/webui/src/static/function.ts]: this function will use in [kill a trial],
I 'll update this function in next commit in PR.
Thanks for reviewing.

File: src/webui/src/static/const.ts (hunk 2 above)
Comment (lvybriage) [reply src/webui/src/static/const.ts]: OK.

File: src/webui/LICENSE [hunk 9]
```diff
@@ -0,0 +1,21 @@
+    MIT License
```
Comment (liuzhe-lz) [inline src/webui/LICENSE]: Is this file necessary?

File: src/webui/src/App.test.tsx [hunk 10]
```diff
@@ -0,0 +1,9 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import App from './App';
+
+it('renders without crashing', () => {
+  const div = document.createElement('div');
```
Comment (liuzhe-lz) [inline src/webui/src/App.test.tsx]: Please keep indentation consistent.

File: src/webui/src/static/function.ts [hunk 11]
```diff
@@ -118,33 +117,36 @@ const killJob = (key: number, id: string, status: string, updateList?: Function)
...
-                message.success('Cancel the job successfully');
+                // message.destroy();
+                // message.success('Cancel the job successfully');
                 // render the table
                 if (updateList) {
                     updateList();  // FIXME
                 }
             } else {
-                message.error('fail to cancel the job');
+                alert('fail to cancel the job');
```
Comment (liuzhe-lz) [inline src/webui/src/static/function.ts]: I believe using browser `alert` API is a bad practice.
We should port to a better notification framework later.

File: src/webui/src/components/overview/SuccessTable.tsx [hunk 12]
```diff
@@ -1,81 +1,128 @@
...
+        });
+        // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+        const newItems = this._copyAndSort(source, currColumn.fieldName!, currColumn.isSortedDescending);
+        this.setState({
+            columns: newColumns,
+            source: newItems
+        });
+    };
+
+    private _copyAndSort<T>(items: T[], columnKey: string, isSortedDescending?: boolean): T[] {
```
Comment (liuzhe-lz) [inline src/webui/src/components/overview/SuccessTable.tsx]: Does TS/React recommend variable name starting with underscore?

File: src/webui/src/components/overview/Progress.tsx [hunk 13]
```diff
@@ -19,24 +19,105 @@ interface ProgressProps {
 
 interface ProgressState {
     isShowLogDrawer: boolean;
+    isCalloutVisible?: boolean;
```
Comment (liuzhe-lz) [inline src/webui/src/components/overview/Progress.tsx]: why `?`?

File: src/webui/src/components/overview/Progress.tsx [hunk 14]
```diff
@@ -19,24 +19,105 @@ interface ProgressProps {
...
+            info, typeInfo,
+            isShowSucceedInfo: true
+        }));
+        setTimeout(this.hideSucceedInfo, 2000);
+    }
+
     editTrialConcurrency = async (userInput: string): Promise<void> => {
         if (!userInput.match(/^[1-9]\d*$/)) {
-            message.error('Please enter a positive integer!', 2);
+            this.getMessageInfo('Please enter a positive integer!', 'error');
```
Comment (liuzhe-lz) [inline src/webui/src/components/overview/Progress.tsx]: I think "_get_ message info" is misleading.

File: src/webui/src/components/overview/Progress.tsx (hunk 13 above)
Comment (lvybriage) [reply src/webui/src/components/overview/Progress.tsx]: when experiment status is `error`,  hover to the status, will tooltip a modal. 
`isCalloutVisible` decide the modal's hidden | show

File: src/webui/src/components/overview/Progress.tsx (hunk 14 above)
Comment (lvybriage) [reply src/webui/src/components/overview/Progress.tsx]: yeah, will change to `showMessageInfo`
It's OK?

File: src/webui/src/components/overview/SuccessTable.tsx (hunk 12 above)
Comment (lvybriage) [reply src/webui/src/components/overview/SuccessTable.tsx]: fabric-react's demos show that, 
but,
将会改成驼峰命名！

File: src/webui/src/static/function.ts (hunk 11 above)
Comment (lvybriage) [reply src/webui/src/static/function.ts]: 你说的对！下一版会fix

File: src/webui/src/App.test.tsx (hunk 10 above)
Comment (lvybriage) [reply src/webui/src/App.test.tsx]: OK.

File: src/webui/LICENSE (hunk 9 above)
Comment (lvybriage) [reply src/webui/LICENSE]: I think so...
